### PR TITLE
Improvements for Picker component

### PR DESCRIPTION
**Pull request by marcinsawicki**
Addresses: https://github.com/livechat/design-system/issues/337

### CODE REVIEW

File: packages/react-components/src/components/Picker/Trigger.tsx [hunk 1]
```diff
@@ -55,9 +57,15 @@ export const Trigger: React.FC<ITriggerProps> = ({
...
+      document.addEventListener('keydown', onKeyDown);
+    }
 
-    return () => document.removeEventListener('keydown', onKeyDown);
+    return () => {
+      if (!disabledSearch) {
+        document.removeEventListener('keydown', onKeyDown);
+      }
+    };
   }, []);
```
Comment (MichalPaszowski) [inline packages/react-components/src/components/Picker/Trigger.tsx:69]: I think 'disabledSearch' should be added to the dependency table. Linter didn't complain about that?

**Review by MichalPaszowski — CHANGES_REQUESTED** [review]

File: packages/react-components/src/components/Picker/PickerList.module.scss [hunk 2]
```diff
@@ -14,6 +14,7 @@ $base-class: 'picker-list';
   padding: 4px 0;
   position: absolute;
   width: 341px;
+  z-index: 10000;
```
Comment (sgraczyk) [inline packages/react-components/src/components/Picker/PickerList.module.scss:17]: Is there any rule on how we use `z-indexes`? Is there any stacking context?

File: packages/react-components/src/components/Picker/Trigger.tsx [hunk 3]
```diff
@@ -14,6 +14,7 @@ const baseClass = 'picker-trigger';
 export type TriggerSize = 'compact' | 'medium' | 'large';
 
 export interface ITriggerProps {
+  disabledSearch: boolean;
```
Comment (sgraczyk) [inline packages/react-components/src/components/Picker/Trigger.tsx:17]: I'd consider alternative naming as in case of `Picker`

Comment (sgraczyk) [reply packages/react-components/src/components/Picker/Trigger.tsx:17]: Plus there is a difference -> `disableSearch` vs `disabledSearch`

File: packages/react-components/src/components/Picker/Picker.tsx [hunk 4]
```diff
@@ -21,7 +21,7 @@ export interface IPickerProps {
   placeholder?: string;
   isRequired?: boolean;
   noSearchResultText?: string;
-  disableSearch?: boolean;
+  searchDisabled?: boolean;
```
Comment (marcinsawicki) [inline packages/react-components/src/components/Picker/Picker.tsx:24]: I've decided to name it `searchDisabled` as this is an interface visible to consumers.
WDYT?

File: packages/react-components/src/components/Picker/Trigger.tsx [hunk 5]
```diff
@@ -14,7 +14,7 @@ const baseClass = 'picker-trigger';
 export type TriggerSize = 'compact' | 'medium' | 'large';
 
 export interface ITriggerProps {
-  disabledSearch: boolean;
+  isSearchDisabled: boolean;
```
Comment (marcinsawicki) [inline packages/react-components/src/components/Picker/Trigger.tsx:17]: Here we have `isSearchDisabled` as this is an internal part of `Picker` component, where our naming convention perfectly fits, the consumer doesn't see it.

File: packages/react-components/src/components/Picker/Picker.tsx (hunk 4 above)
Comment (sgraczyk) [reply packages/react-components/src/components/Picker/Picker.tsx:24]: Fine for now

File: packages/react-components/src/components/Picker/Trigger.tsx (hunk 5 above)
Comment (sgraczyk) [reply packages/react-components/src/components/Picker/Trigger.tsx:17]: Fine for now, we'll rethink this later

File: packages/react-components/src/utils/StackingContextLevel.scss [hunk 6]
```diff
@@ -0,0 +1,5 @@
+/**
+ * Definition of z-index values to be used across application.
+ */
+
+$stacking-context-level-dropdown: 10000;
```
[the remark this answers is not in the document]
Comment (sgraczyk) [reply packages/react-components/src/utils/StackingContextLevel.scss:5]: It is ok now, however, I was wondering if 1000 would be enough.

**Review by sgraczyk — APPROVED** [review]

**Review by MichalPaszowski — APPROVED** [review]

**Pull request merged**